certs/bootstrap/certs.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package bootstrap
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mainflux/mainflux/certs"
+	"github.com/mainflux/mainflux/pkg/errors"
+
+	"os"
+	"sync"
+
+	mfsdk "github.com/mainflux/mainflux/pkg/sdk/go"
+)
+
+var (
+	// ErrUnauthorizedAccess indicates missing or invalid credentials provided
+	// when accessing a protected resource.
+	ErrUnauthorizedAccess = errors.New("unauthorized access to Bootstrap service")
+
+	// ErrUnexpectedBSResponse indicates unexpected response from Bootstrap service.
+	ErrUnexpectedBSResponse = errors.New("unexpected Bootstrap service response")
+)
+
+type bootstrapClient struct {
+	updateURL string
+	token     string
+	email     string
+	pass      string
+	sdk       mfsdk.SDK
+	mu        sync.Mutex
+}
+
+// New returns new Bootstrap service client
+func New(updateURL, email, pass string, sdk mfsdk.SDK) certs.BootstrapClient {
+	token := os.Getenv("MF_USERS_TOKEN")
+
+	return &bootstrapClient{
+		updateURL: updateURL,
+		email:     email,
+		pass:      pass,
+		sdk:       sdk,
+		token:     token,
+		mu:        sync.Mutex{},
+	}
+}
+
+func (c *bootstrapClient) fetchToken() string {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	return c.token
+}
+
+func (c *bootstrapClient) login() (string, error) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	user := mfsdk.User{Email: c.email, Password: c.pass}
+	token, err := c.sdk.CreateToken(user)
+	if err != nil {
+		return "", err
+	}
+	c.token = token
+
+	return c.token, nil
+}
+
+func (c *bootstrapClient) UpdateCerts(ctx context.Context, thingID, clientCert, clientKey, caCert string) error {
```

Comment:
Why don't we use `c.sdk.UpdateBootstrapCerts` here?

Reply:
Okay I will try to use c.sdk.UpdateBootstrapCerts

Reply:
@dborovcanin, we can't use c.sdk.UpdateBootstrapCerts , since SDK simply reply with status , and we need more detailed about the response body 
https://github.com/mainflux/mainflux/blob/d73a5d53fec0313a51ca933321ea75de7ad0c90e/pkg/sdk/go/bootstrap.go#L180

Reply:
> @dborovcanin, we can't use c.sdk.UpdateBootstrapCerts , since SDK simply reply with status , and we need more detailed about the response body
> 
> https://github.com/mainflux/mainflux/blob/d73a5d53fec0313a51ca933321ea75de7ad0c90e/pkg/sdk/go/bootstrap.go#L180

Is this only because SDK does not return return error based on the reponse, but on status code? In that case, we should update SDK to return more specifc error.

Reply:
What about this?

Reply:
yes it use sdk now